pandas/core/arrays/arrow/dtype.py

```diff
@@ -59,29 +75,44 @@ def construct_from_string(cls, string: str):
         Parameters
         ----------
         string : str
+            string should follow the format f"{pyarrow_type}[pyarrow]"
+            e.g. int64[pyarrow]
```

Comment:
This pattern might be a bit problematic (or at least strange looking) for arrow types that are itself parametrized? (eg `"timestamp[s]"`)

Reply:
Actually I suppose the implementation below also doesn't work for those cases.

Reply:
Ah yeah this is probably not robust to the parameterized arrow dtypes. Maybe for those parameterized types we raise and tell users to build a `ArrowDtype` because maintaining parsing logic or a mapping doesn't sound fun.

Reply:
Hmm, on the other hand, this is _also_ an issue for the type repr itself?

```
In [2]: ArrowDtype(pa.timestamp("s"))
Out[2]: timestamp[s][pyarrow]
```

Also here it looks a bit strange I would say, although that is maybe fine? 

The `DatetimeTZDtype.construct_from_string` is an example that uses a regular expression to get the timezone from the parametrized string repr. That could be prior art for here as well (but certainly doesn't need to be in this PR, raising for now sounds good).